config/routes.rb

```diff
@@ -36,6 +36,14 @@
 
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
+  namespace :hyrax, path: :concern do
+    namespaced_resources 'datasets' do
+      member do
+        get :assign_register_doi
```

Comment:
Given that this is a request to modify the resource, I would prefer `post` as the method here, but wouldn't insist on it if folks think that's too pedantic.